experimental/dds/tree2/src/feature-libraries/simple-tree/proxyNodeCursor.ts

```diff
@@ -0,0 +1,193 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { assert } from "@fluidframework/core-utils";
+
+import {
+	EmptyKey,
+	rootFieldKey,
+	type FieldKey,
+	type TreeTypeSet,
+	type Value,
+	CursorLocationType,
+} from "../../core";
+// eslint-disable-next-line import/no-internal-modules
+import { leaf } from "../../domains/leafDomain";
+import {
+	type ContextuallyTypedNodeData,
+	type TreeDataContext,
+	getPossibleTypes,
+	isFluidHandle,
+	isPrimitiveValue,
+} from "../contextuallyTyped";
+import {
+	type CursorAdapter,
+	type CursorWithNode,
+	stackTreeNodeCursor,
+	stackTreeFieldCursor,
+} from "../treeCursorUtils";
+import { TreeNodeSchema } from "../typed-schema";
+import { type ProxyNode } from "./types";
+/**
+ * Creates a {@link CursorAdapter} capable of mapping proxy-domain trees to the editable tree domain.
+ *
+ * @param context - The {@link TreeEntity.context} of the parent node under which the tree is being inserted.
+ * @param typeSet - The types allowed under the parent node under which the tree is being inserted.
+ *
+ * @privateRemarks
+ * This is exported solely for testing purposes.
+ * Consumers should use {@link cursorFromProxyTree} instead.
+ */
+export function createProxyTreeAdapter<TNode extends ProxyNode<TreeNodeSchema, "javaScript">>(
+	context: TreeDataContext,
+	typeSet: TreeTypeSet,
+): CursorAdapter<TNode> {
+	return {
+		value: (node: TNode) => {
+			return isPrimitiveValue(node) || node === null || isFluidHandle(node)
+				? (node as Value)
```

Comment:
I'm surprised this cast is needed. Can it be removed? If now, why?

Reply:
It was at one point in time, but the code has been shuffled significantly since. I'll verify and remove if possible,